### PR TITLE
Cohesive networks

### PR DESCRIPTION
**Pull request by HKF1977**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by HKF1977** [comment]
Hi, I'm struggling to get past "Error: Code=MarketplacePurchaseEligibilityFailed; " I understand the problem, but can't see how I can accept the legal terms in the pipelines subscription? Any help is much appreciated. Thanks

**Comment by HKF1977** [comment]
@bmoore-msft Hi, can you give me some pointers on the above^, I've updated the terms in my Cohesive Azure account for the Cohesive subscription. That didn't solve it, my read is the pipeline account needs to agree the terms?  Is it something I need to solve or does it require some intervention? Thanks Scott.

File: cohesive-vns3-free-try-now/metadata.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+{
+    "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+    "itemDisplayName": "VNS3 network appliance for cloud connectivity and security.",
+    "description": "VNS3 is a software only virtual appliance that provides the combined features and functions of a Security Appliance, Application Delivery Controller and Unified Threat Management device at the cloud application edge. Key benefits, On top of cloud networking, Always on end to end encryption, Federate data centres, cloud regions, cloud providers, and/or containers, creating one unified address space, Attestable control over encryption keys, Meshed network manageable at scale, Reliable HA in the Cloud, Isolate sensitive applications (fast low cost Network Segmentation), Segmentation within applications, Analysis of all data in motion in the cloud. Key network functions; virtual router, switch, firewall, vpn concentrator, multicast distributor, with plugins for WAF, NIDS, Caching, Proxy Load Balancers and other Layer 4 thru 7 network functions, VNS3 doesn't require new knowledge or training to implement, so you can integrate with existing network equipment.",
```
Comment (bmoore-msft) [inline cohesive-vns3-free-try-now/metadata.json:5]: fix typos (e.g. capitalization after commas)

File: cohesive-vns3-free-try-now/metadata.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    "summary": "VNS3 network appliance for cloud connectivity,federation and security.",
```
Comment (bmoore-msft) [inline cohesive-vns3-free-try-now/metadata.json:6]: add space after comma

File: cohesive-vns3-free-try-now/azuredeploy.parameters.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "adminUsername": {
+            "value": "GEN-UNIQUE"
+        },
+        "adminPassword": {
+            "value": "GEN-PASSWORD"
+        },
+        "numberOfInstances": {
```
Comment (bmoore-msft) [inline cohesive-vns3-free-try-now/azuredeploy.parameters.json:11]: add a defaultValue for this and remove from here

File: cohesive-vns3-free-try-now/azuredeploy.parameters.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+            "value": 3
+        },
+        "virtualMachineOs": {
```
Comment (bmoore-msft) [inline cohesive-vns3-free-try-now/azuredeploy.parameters.json:14]: remove this and use the defaultValue

File: cohesive-vns3-free-try-now/README.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+[![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fcohesive-vns3-free-try-now%2Fazuredeploy.json)
+
+### Introduction
+This Quickstart automates the process of deploying a Cohesive Networks VNS3 overlay network in Azure.
+
+It’s intended for users who are interested in a generalised method for encrypting all data in motion between Azure virtual machines, connecting Azure VNETs securely to other networks, and securing traffic with firewalling and network function virtualisation. VNS3 allows you to meet compliance requirements, attest to data security, and manage your cloud deployments. The VNS3 overlay federates resources for multi-region, hybrid cloud and multi-cloud deployments
```
Comment (bmoore-msft) [inline cohesive-vns3-free-try-now/README.md:19]: add period after the last sentence

File: cohesive-vns3-free-try-now/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,629 @@
...
+            }
+        },
+        {
+            "type": "Microsoft.Network/virtualNetworks/subnets",
+            "apiVersion": "2019-06-01",
+            "name": "[concat(variables('virtualNetworkName'), '/', variables('publicSubnet'))]",
+            "properties": {
+                "addressPrefix": "[variables('vnetPublicSubnet')]"
+            },
+            "dependsOn": [
```
Comment (bmoore-msft) [inline cohesive-vns3-free-try-now/azuredeploy.json:420]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#sort-order-of-properties for the proper sort order of properties

File: cohesive-vns3-free-try-now/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,629 @@
...
+            "type": "Microsoft.Network/virtualNetworks/subnets",
+            "apiVersion": "2019-06-01",
+            "name": "[concat(variables('virtualNetworkName'), '/', variables('privateSubnet'))]",
+            "properties": {
+                "addressPrefix": "[variables('vnetPrivateSubnet')]",
+                "routeTable": {
+                    "id": "[resourceId('Microsoft.Network/routeTables', variables('routeTableName'))]"
+                }
+            },
+            "dependsOn": [
```
Comment (bmoore-msft) [inline cohesive-vns3-free-try-now/azuredeploy.json:434]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#sort-order-of-properties for the proper sort order of properties

File: cohesive-vns3-free-try-now/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,629 @@
...
+                            "addressPrefix": "0.0.0.0/0",
+                            "nextHopType": "VirtualAppliance",
+                            "nextHopIpAddress": "10.10.10.10",
+                            "hasBgpOverride": false
+                        }
+                    }
+                ]
+            }
+        },
+            {
```
Comment (bmoore-msft) [inline cohesive-vns3-free-try-now/azuredeploy.json:458]: format the json, e.g. SHIFT+ALT+F in VS Code

**Review by bmoore-msft — CHANGES_REQUESTED** [review]
re: the eligibility check - what (and how) needs to be set in the subscription to enable purchase of the offer?

**Comment by HKF1977** [comment]
re: the eligibility check - From the error and suggested solution. It looks to be the marketplace terms. 

Here is the Powershell command: Get-AzureRmMarketplaceTerms -Publisher "cohesive" -Product "cohesive-vns3-5_0_x" -Name "cohesive-vns3-5_0_x-free" | Set-AzureRmMarketplaceTerms -Accept. 

Ref. this post: https://techgenix.com/azure-marketplace-purchase-eligibility-failed-fix/

thx

File: cohesive-vns3-free-try-now/azuredeploy.json [hunk 9]
```diff
@@ -1,629 +1,628 @@
...
+  "parameters": {
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "User Name for VM's (required but not used for VNS3)"
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "defaultValue": "GEN-PASSWORD",
```
Comment (bmoore-msft) [inline cohesive-vns3-free-try-now/azuredeploy.json:13]: the "GEN" tokens only work in the parameter file, remove the defaultValue and move this back to the param file

File: cohesive-vns3-free-try-now/README.md [hunk 10]
```diff
@@ -16,7 +16,7 @@
 ### Introduction
 This Quickstart automates the process of deploying a Cohesive Networks VNS3 overlay network in Azure.
 
-It’s intended for users who are interested in a generalised method for encrypting all data in motion between Azure virtual machines, connecting Azure VNETs securely to other networks, and securing traffic with firewalling and network function virtualisation. VNS3 allows you to meet compliance requirements, attest to data security, and manage your cloud deployments. The VNS3 overlay federates resources for multi-region, hybrid cloud and multi-cloud deployments
+It’s intended for users who are interested in a generalised method for encrypting all data in motion between Azure virtual machines, connecting Azure VNETs securely to other networks, and securing traffic with firewalling and network function virtualisation. VNS3 allows you to meet compliance requirements, attest to data security, and manage your cloud deployments. The VNS3 overlay federates resources for multi-region, hybrid cloud and multi-cloud deployments.
```
Comment (bmoore-msft) [inline cohesive-vns3-free-try-now/README.md:19]: add a comment/section somewhere in the readme (doesn't have to be here) about accepting the marketplace terms, e.g.

```
Get-AzMarketplaceTerms -Publisher "cohesive" -Product "cohesive-vns3-5_0_x" -Name "cohesive-vns3-5_0_x-free" | Set-AzMarketplaceTerms -Accept
```

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Pull request merged**